### ISSUE TITLE
Visual Studio Code plugin does not create a top-level ios folder

### ISSUE THREAD

**Issue by davidair** [flutter/flutter#22772]
## Steps to Reproduce

1. Open Visual Studio Code 1.27.2
2. View | Command Palette | Flutter: New Project
3. Call the project "foo", save it in some folder

Expected: a top-level ios folder to be created (such as when a Flutter project is created in Android Studio)
Actual: no ios folder is created (although one gets generated under build/ios after running the app)

```
[✓] Flutter (Channel master, v0.9.7-pre.61, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.9.7-pre.61 at /Users/davidair/github/flutter/flutter
    • Framework revision 2d81adf74c (20 hours ago), 2018-10-05 22:29:37 -0700
    • Engine revision 572fa5646a
    • Dart version 2.1.0-dev.6.0.flutter-c6254163dc

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/davidair/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version openjdk version "1.8.0_152-release"
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode9.2.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.0

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 25.0.1
    • Dart plugin version 173.4700
    • Java version openjdk version "1.8.0_152-release"

[✓] IntelliJ IDEA Community Edition (version 2018.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 24.2.2
    • Dart plugin version 181.4203.498

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.19.0

[✓] Connected device (1 available)
    • iPhone X • 4A159F1C-EFDD-4FD0-957B-01BE7573C62C • ios • iOS 11.2 (simulator)

• No issues found!
```

**Comment by devoncarew** [comment]
@davidair, is a hidden (`.ios/`) folder created?

**Comment by devoncarew** [comment]
Gotcha; that's expected now, and part of a recent change to the default project structure.

@matthew-carroll - this is likely something we'll want to add to the FAQ, and / or email out to flutter-dev to give people a heads up.

**Comment by davidair** [comment]
There are a couple of issues here:

1. It looks like the .ios folder gets automatically regenerated each time the pubspec.yaml changes. This breaks any workflows that require adding files to the Xcode project, including the Firebase workflow: https://firebase.google.com/docs/flutter/setup#configure_ios

2. In Android Studio (3.2, with the latest plugins), I am still seeing the ios folder being created, so it's inconsistent with the VS Code experience

**Comment by devoncarew** [comment]
> It looks like the .ios folder gets automatically regenerated each time the pubspec.yaml changes. 

I think this is unexpected; the project should only be regenerated if you re-run `flutter create`. Can you open a new issue for this (and cc @matthew-carroll and @gspencergoog)? Thanks!

> In Android Studio (3.2, with the latest plugins), I am still seeing the ios folder being created, so it's inconsistent with the VS Code experience

Android Studio is using the (now older) `app` project template. Once the `application` template is available in the beta channel, we can switch Android Studio over.

**Comment by matthew-carroll** [comment]
@davidair Thanks for pointing out the plugin issue.  I've filed the following ticket to track that and I expect that we'll have a fix out within the next couple of weeks:
https://github.com/flutter/flutter/issues/22832

The following command was just merged to master and it will turn your generated host apps to editable host apps to get around this problem for now:
`flutter make-host-app-editable`

@devoncarew the regeneration is expected.  The reason we regenerate is because the underlying project structure could change at any time.  For example, Android or iOS could add/remove project directories, manifest/plist properties, or Gradle/XCode configuration.  We want these changes to be transparent to developers and therefore the generated projects should be considered completely ephemeral.  The core problem here is that we failed to expose common configuration controls for the ephemeral project.
